src/App.js

```diff
@@ -1,11 +1,93 @@
 import React from 'react';
-import './App.css';
+import {getAllTodos} from './components/util/TodoApi';
+import {getAllUsers} from './components/util/UserApi';
+import TodoList, {
+    SORT_ORDER_COMPLETED, SORT_ORDER_TITLE, SORT_ORDER_USER
+} from './components/todo-list/TodoList';
 
-function App() {
-  return (
-    <div className="App">
-    </div>
-  );
+
+export default class App extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            sortField: SORT_ORDER_TITLE,
+            isLoading: false,
+            todoData: [],
+        };
+
+        this.handleSort = (sortField) => {
+            this.setState(() => {
+                return {
+                    sortField: sortField
+                }
+            });
+        };
+    }
+
+    loadData = async () => {
+        this.setState({isLoading: true});
+        const [todos, users] = await Promise.all([
+            getAllTodos(),
+            getAllUsers(),
+        ]);
+        const items = this.getTodosWithUsers(todos, users);
+        this.setState({
+            todosData: items,
+            isLoading: false
+        });
+    };
+
+    getTodosWithUsers = (todos, users) => {
+        return todos.map(todo => {
+            return {
+                ...todo,
+                user: users.find(user => user.id === todo.userId)
+            };
+        });
+    };
+
+    sortTodos(todos, sortField) {
+        let callbackMap = {
+            [SORT_ORDER_TITLE.name]: (a, b) => a.title.localeCompare(b.title),
+            [SORT_ORDER_USER.name]: (a, b) => a.user.name.localeCompare(b.user.name),
+            [SORT_ORDER_COMPLETED.name]: (a, b) => a.completed - b.completed,
+        };
+
+        const callback = callbackMap[sortField.name] || callbackMap.SORT_ORDER_TITLE;
+        if (todos) {
+            if (sortField.order) {
+                sortField.order = !sortField.order;
+                return todos.sort(callback);
+            } else {
+                sortField.order = !sortField.order;
+                return todos.sort(callback).reverse();
```

Comment:
If you add local variable `order`, you can get rid of `reverse()` and nested `if (condition1) { if (condition2) {}}`
It can help you avoid unneeded array processing.